src/posix/platform/radio_spinel.hpp

```diff
@@ -603,8 +603,7 @@ class RadioSpinel : public HdlcInterface::Callbacks
      */
     bool IsSafeToHandleNow(spinel_prop_key_t aKey) const
     {
-        return !((mHdlcInterface.IsDecoding() || mWaitingKey != SPINEL_PROP_LAST_STATUS) &&
-                 (aKey == SPINEL_PROP_STREAM_RAW || aKey == SPINEL_PROP_MAC_ENERGY_SCAN_RESULT));
+        return !(aKey == SPINEL_PROP_STREAM_RAW || aKey == SPINEL_PROP_MAC_ENERGY_SCAN_RESULT);
```

Comment:
Is it better to keep `mWaitingKey != SPINEL_PROP_LAST_STATUS`?

Reply:
To keep a line with the original logic of the code, we don't need to keep `mWaitingKey != SPINEL_PROP_LAST_STATUS`. @abtink How do you think about it?

Reply:
@zhanglongxia , I agree we should not need it.

As you described in the commit log (I also read through the chain and verified it), whenever we get to `IsSafeToHandle()` the `IsDecoding()` should always return `true`. 

So 
`(mHdlcInterface.IsDecoding() || mWaitingKey != SPINEL_PROP_LAST_STATUS)` is always treated as `(true || mWaitingKey != SPINEL_PROP_LAST_STATUS)` which should become `true` (which is then `&&` with other checks).

Let me add the flow of calls:
- `IsSafeToHandleNow()` is only called from `HandleNotification(RxFrameBuffer)`.
- `HandleNotification(RxFrameBuffer)` is called from `HandleSpinelFrame()`
- This  itself is called from `HdlcInterface::Callbacks::HandleReceivedFrame()`
- The HDLC callback can be invoked when HDLC decoder `mHdlcDecoder` finishes decoding a full frame.
- This can only happen when `mHdlcDecoder.Decode()` is called and right before this we set `mIsDecoding` to `true`.

I think the reason we had this check in `IsSafeToHandle()` is that in earlier version `HandleNotification(RxFrameBuffer)` could be called to process queued frames. But in current master code we have two `HandleNotification()` methods (one with `RxFrameBuffer` and other one with a frame pointer and length), and the second is used for queued frames.

Reply:
I think when `mWaitingKey == SPINEL_PROP_LAST_STATUS`, it's safe to handle now, for performance consideration, I prefer keep the check for `mWaitingKey`. Thoughts?

Reply:
@bukepo , I agree it should be fine to add such a check (to handle such frame more quickly instead of saving them in queue and processing them later).  

It would change the behavior of code though (as in the existing code we were never actually using this path in the code). This may change the order of processing of some of the received frames. It sounds ok/fine to me.

Reply:
I think we may have a typo/error in the existing code and the `||` should have been `&&` in 
`(mHdlcInterface.IsDecoding() || mWaitingKey != SPINEL_PROP_LAST_STATUS)`.

The logic behind this is "if HDLC is decoding but we are not waiting for a response (indicated by `mWaitingKey` being `LAST_STATUS`), it is safe to handle", or saying it other way: "it is NOT safe to handle if HDLC is decoding AND we are waiting for a response.

Does it sound reasonable?

Reply:
@abtink Thanks for your detailed explanation. I have updated the code.